packages/web3-eth-accounts/src/errors.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable max-classes-per-file */
+
+export abstract class Web3Error extends Error {
+	public readonly name: string;
+
+	public constructor(value: unknown, msg: string) {
+		super(`Invalid value given "${String(value)}". Error: ${msg}.`);
+		this.name = this.constructor.name;
+		Error.captureStackTrace(this, Web3Error);
+	}
+
+	public toJSON() {
+		return { name: this.name, message: this.message };
+	}
+}
```

Comment:
Should be imported from `web3-common` instead of being re-declared (will help us remain consistent if we ever make changes to `Web3Error`)